keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -172,27 +172,50 @@ type PersistentVolumeClaimSpec struct {
   ...
 }
 
+// Add the PersistentVolumeClaimModifyVolume condition to existing PersistentVolumeClaimConditionType
+// Condition is used to document the last error controller sees
+const (
+  ...
+  // PersistentVolumeClaimModifyVolume - a user trigger modify volume of pvc has been started
+  PersistentVolumeClaimModifyVolume PersistentVolumeClaimConditionType = "ModifyVolume"
+  ...
+)
+
+
 // PersistentVolumeClaimStatus represents the status of PV claim
 type PersistentVolumeClaimStatus struct {
   ...
-  // The VolumeAttributesClassName string cannot be empty
-  VolumeAttributesClassName string
-  ModifyVolumeStatus *PersistentVolumeClaimModifyVolumeStatus
+  // ModifyVolumeStatus represents the status object of ControllerModifyVolume operation
+  ModifyVolumeStatus ModifyVolumeStatus
   ...
 }
 
+// ModifyVolumeStatus represents the status object of ControllerModifyVolume operation
+type ModifyVolumeStatus struct {
+  // ActualVolumeAttributesClassName is the actual name of the VolumeAttributesClass the PVC is using
+  ActualVolumeAttributesClassName string
+  // TargetVolumeAttributesClassName is the target name of the VolumeAttributesClass the PVC is using
+  TargetVolumeAttributesClassName string
+  // Status is the status of the ControllerModifyVolume operation
+  Status *PersistentVolumeClaimModifyVolumeStatus
+}
+
 // +enum
+// When a controller receives persistentvolume claim update with PersistentVolumeClaimModifyVolumeStatus for a resource
+// that it does not recognizes, then it should ignore that update and let other controllers
+// handle it.
 type PersistentVolumeClaimModifyVolumeStatus string
 
 const (
-  // When modify volume is complete, the empty string is set by modify volume controller.
-  PersistentVolumeClaimNoModifyVolumeInProgress PersistentVolumeClaimModifyVolumeStatus = ""
+  // State set when modify volume controller wants to modify the volume in control-plane
```

Comment:
What is the `modify volume controller`? pv controller in kube-controller-manager or CSI driver controller?

Reply:
it is the resize controller actually, we are putting this modify volume controller logic in the external-resizer.